packages/terraform-modules-public/gcp/example/main.tf

```diff
@@ -115,3 +116,110 @@ module "celo_cluster" {
+  for_each = var.stackdriver_logging_exclusions
+  
+  name            = each.key                    #maybe make this a random_id to ensure no naming conflicts
+  description     = each.value["description"]
+  filter          = each.value["filter"]
+}
+
+resource "random_id" "stackdriver_logging_exclusions" {
+  for_each = var.stackdriver_logging_exclusions
+    byte_length = 4
+}
+
+resource "random_id" "stackdriver_logging_metrics" {
+  for_each = var.stackdriver_logging_metrics
+    byte_length = 4
+}
+
+resource "google_logging_metric" "logging_metric" {
+  for_each = var.stackdriver_logging_metrics
+    name        = each.key
+    description = each.value["description"]
+    filter = each.value["filter"]
+    metric_descriptor {
+      metric_kind  = "DELTA"
+      value_type   = "INT64"
+      display_name = each.value["description"]
+    }
+}
+
+resource "google_logging_metric" "distribution_blocks_ingested" {
+  name   = "tf_eth_blocks_ingested"
+  description = "Ethereum blocks ingested"
+  filter = "resource.type=\"gce_instance\" AND \"Imported new chain segment\""
+  metric_descriptor {
+    metric_kind = "DELTA"
+    value_type  = "DISTRIBUTION"
+    unit        = "blocks"
+    display_name = "Blocks Ingested"
+  }
+  value_extractor = "REGEXP_EXTRACT(jsonPayload.message, \"\\\"blocks\\\":(\\\\d+)\")"
+  bucket_options {
+    explicit_buckets {
+      bounds = [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,40,60,80,100,120,140,160,180,200,400,500,600,700,800,900,1000,1200,1400,1600,1800,2000,2200,2400,2600,2800,3000,3500,4000,5000]
+    }
+  }
+}
+
+resource "google_storage_bucket" "chaindata_bucket" {
+  name = "${var.google["project"]}-chaindata"
+  location = "US"
+
+  lifecycle_rule {
+    condition {
+      num_newer_versions = 2
+    }
+    action {
+      type = "Delete"
+    }
+  }
+
+  versioning {
+      enabled = true
+    }
+}
+
+resource "google_storage_bucket_iam_binding" "chaindata_binding_write" {
+  bucket = "${var.google["project"]}-chaindata"
+  role = "roles/storage.objectCreator"
+  members = [
+    "serviceAccount:${var.GCP_DEFAULT_SERVICE_ACCOUNT}",
+  ]
+}
+
+resource "google_storage_bucket_iam_binding" "chaindata_binding_read" {
+  bucket = "${var.google["project"]}-chaindata"
+  role = "roles/storage.objectViewer"
+  members = [
+    "serviceAccount:${var.GCP_DEFAULT_SERVICE_ACCOUNT}",
+  ]
+}
+
+#resource "google_storage_bucket" "public_www_bucket" {
+#  name = var.public_www_fqdn
+#  location = "US"
+#  force_destroy = true
+#
+#  website {
+#    main_page_suffix = "index.html"
+#    not_found_page   = "404.html"
+#  }
+#  cors {
+#    origin          = ["https://${var.public_www_fqdn}"]
+#    method          = ["GET", "HEAD"]
+#    response_header = ["*"]
+#    max_age_seconds = 3600
+#  }
+#}
+
+#resource "google_storage_bucket_iam_binding" "public_www_binding_read" {
+#  bucket = var.public_www_fqdn
+#  role = "roles/storage.objectViewer"
+#  members = [
+#    "allUsers"
+#  ]
+#}
```

Comment:
Should this be removed?

Reply:
No.  Validators need to host metadata on a publicly accessible web server, so I enabled this to make it easy to do so on GCS.  Agree it was confusing having it in there but commented so, so it's now enabled.

https://docs.celo.org/celo-codebase/protocol/identity/metadata